aiozk/client.py

```diff
@@ -25,7 +25,6 @@ def __init__(
         retry_policy=None,
         allow_read_only=False,
         read_timeout=None,
-        loop=None,
```

Comment:
TL;DR IMO it's OK to remove all loop parameters from aiozk

I worried about the use case of mixing event loops by aiozk users. Especially the case of combining multiple event loops while handling one ZKClient instance.

But aiozk was already not working with multiple event loops. See this example below.

```
def test_multiple_loops(path):
    with pytest.raises(RuntimeError):
        asyncio.get_running_loop()

    loop1 = asyncio.new_event_loop()
    zk = get_client(loop1)
    loop1.run_until_complete(zk.start())
    loop1.run_until_complete(zk.create(path, 'data'))

    loop2 = asyncio.new_event_loop()
    loop2.run_until_complete(zk.get_data(path))
```

When the test code is run, the error occurs:

```
    async def send(self, request):
        response = None
        while not response:
            await self.retry_policy.enforce(request)
            await self.ensure_safe_state(writing=request.writes_data)

            try:
                self.xid += 1
                if self.xid > 0x7fffffff:
                    # xid should not exceed the maximum of 32 bit signed integer
                    # and it should be positive value because a few negative
                    # values are special xid.
                    self.xid = 1
>               zxid, response = await self.conn.send(request, xid=self.xid)

E   RuntimeError: Task <Task pending name='Task-10' coro=<ZKClient.get_data() running at /home/esrse/opensource/aiozk/aiozk/client.py:318> cb=[_run_until_complete_cb() at /home/esrse/.pyenv/versions/3.8.6/lib/python3.8/asyncio/base_events.py:184]> got Future <Future pending> attached to a different loop
```

And the line of test code causing the error is:
```
loop2.run_until_complete(zk.get_data(path))
```

This means that the future returned by `self.conn.send()` is attached to a different loop (which is referred by `self.loop`) but the current loop which is different from `self.loop` is trying `await`ing the future.

It implies that even though we uphold `self.loop` inside aiozk, we should fail when we try `await`ing, because `self.loop` and the current loop which deals with `await` are different.

A ZKClient instance from a event loop should be used always with the same event loop until it is destroyed.


The new code patched by this PR also fails with combined event loops but it leads to different behavior.

`RuntimeError` does not occur, but the test code never completes but hangs forever at this line:
```
loop2.run_until_complete(zk.get_data(path))
```


I can't find ordinary working examples of multiple event loops with one ZKClient instance which upholds self.loop attributes. Thus I conclude that it's OK to remove `self.loop`. But it doesn't mean that it's impossible to use multiple event loops for one ZKClient instance, but I signify that using different event loops for one ZKClient instance is a pretty bad pattern so we should not do that. And also doing that will incur lots of undefined behaviors.